solver/src/solver/http_solver.rs

```diff
@@ -144,86 +137,87 @@ impl HttpSolver {
             .collect()
     }
 
-    fn map_orders_for_solver(&self, orders: Vec<LimitOrder>) -> HashMap<String, LimitOrder> {
+    fn map_orders_for_solver(&self, orders: Vec<LimitOrder>) -> HashMap<usize, LimitOrder> {
         orders
             .into_iter()
             .enumerate()
-            .map(|(index, order)| (index.to_string(), order))
+            .map(|(index, order)| (index, order))
             .collect()
     }
 
     async fn order_models(
         &self,
-        orders: &HashMap<String, LimitOrder>,
+        orders: &HashMap<usize, LimitOrder>,
         gas_price: f64,
-    ) -> Result<HashMap<String, OrderModel>> {
+    ) -> Result<HashMap<usize, OrderModel>> {
         let order_cost = self.order_cost(gas_price).await;
-        let mut result: HashMap<String, OrderModel> = HashMap::new();
+        let mut result: HashMap<usize, OrderModel> = HashMap::new();
         for (index, order) in orders {
-            let order_fee = self.order_fee(&order)?;
+            let order_fee = self.order_fee(&order);
             let order = OrderModel {
-                sell_token: self.token_to_string(&order.sell_token),
-                buy_token: self.token_to_string(&order.buy_token),
+                sell_token: order.sell_token,
+                buy_token: order.buy_token,
                 sell_amount: order.sell_amount,
                 buy_amount: order.buy_amount,
                 allow_partial_fill: order.partially_fillable,
                 is_sell_order: matches!(order.kind, OrderKind::Sell),
                 fee: FeeModel {
                     amount: order_fee,
-                    token: self.token_to_string(&order.sell_token),
+                    token: order.sell_token,
                 },
                 cost: CostModel {
                     amount: order_cost,
-                    token: self.token_to_string(&self.native_token),
+                    token: self.native_token,
                 },
             };
-            result.insert(index.clone(), order);
+            result.insert(*index, order);
         }
         Ok(result)
     }
 
-    fn map_amms_for_solver(&self, orders: Vec<AmmOrder>) -> HashMap<String, AmmOrder> {
+    fn map_amms_for_solver(&self, orders: Vec<AmmOrder>) -> HashMap<usize, AmmOrder> {
         orders
             .into_iter()
             .enumerate()
-            .map(|(index, amm)| (index.to_string(), amm))
+            .map(|(index, amm)| (index, amm))
             .collect()
     }
 
     async fn amm_models(
```

Comment:
If `uniswap_cost` is no longer `async` then I think this method doesn't have to be `async` either.